cpp/oneapi/dal/backend/primitives/sort/sort_dpc.cpp

```diff
@@ -0,0 +1,540 @@
+            const std::uint32_t sub_group_id = sbg.get_group_id();
+            const std::uint32_t group_id = item.get_group().get_id(0) * n_sub_groups + sub_group_id;
+
+            IndexType ind_start = group_id * elems_for_sbg;
+            IndexType ind_end = (group_id + 1) * elems_for_sbg;
+
+            if (ind_end > elem_count) {
+                ind_end = elem_count;
+            }
+
+            IndexType offset[(std::uint32_t)1 << radix_bits];
+            const std::uint32_t radix_range = (std::uint32_t)1 << radix_bits;
+            const std::uint32_t radix_range_1 = radix_range - 1;
+            for (std::uint32_t i = 0; i < radix_range; i++) {
+                offset[i] = 0;
+            }
+
+            for (IndexType i = ind_start + local_id; i < ind_end; i += local_size) {
+                RadixIntType data_bits = ((inv_bits(val_ptr[i]) >> bit_offset) & radix_range_1);
+                for (std::uint32_t j = 0; j < radix_range; j++) {
+                    IndexType value = static_cast<IndexType>(data_bits == j);
+                    IndexType partial_offset =
+                        sycl::ONEAPI::reduce(sbg, value, cl::sycl::ONEAPI::plus<IndexType>());
+                    offset[j] += partial_offset;
+                }
+            }
+
+            if (local_id == 0) {
+                for (std::uint32_t j = 0; j < radix_range; j++) {
+                    part_hist_ptr[group_id * radix_range + j] = offset[j];
+                }
+            }
+        });
+    });
+
+    return event;
+}
+
+template <typename RadixIntType, typename IndexType>
+static sycl::event radix_hist_scan(sycl::queue& queue,
+                                   const ndarray<IndexType, 1>& part_hist,
+                                   ndarray<IndexType, 1>& part_prefix_hist,
+                                   std::int64_t local_size,
+                                   std::int64_t local_hist_count) {
+    ONEDAL_ASSERT(part_hist.get_count() == ((local_hist_count + 1) << radix_bits));
+    ONEDAL_ASSERT(part_prefix_hist.get_count() == ((local_hist_count + 1) << radix_bits));
+
+    const IndexType* part_hist_ptr = part_hist.get_data();
+    IndexType* part_prefix_hist_ptr = part_prefix_hist.get_mutable_data();
+
+    sycl::range<1> global(local_size);
+    sycl::range<1> local(local_size);
+    sycl::nd_range<1> nd_range(global, local);
+
+    auto event = queue.submit([&](cl::sycl::handler& cgh) {
+        cgh.parallel_for(nd_range, [=](sycl::nd_item<1> item) {
+            auto sbg = item.get_sub_group();
+            if (sbg.get_group_id()[0] > 0) {
+                return;
+            }
+
+            const std::uint32_t local_size = sbg.get_local_range()[0];
+            const std::uint32_t local_id = sbg.get_local_id()[0];
+
+            IndexType offset[(std::uint32_t)1 << radix_bits];
+            const std::uint32_t radix_range = (std::uint32_t)1 << radix_bits;
```

Comment:
This line may be pulled up before the kernel and be `constexpr`, then you can use it here:
```c++
IndexType offset[radix_range];
```

Reply:
fixed